components/discord_bot/actions/send-message-with-file/send-message-with-file.mjs

```diff
@@ -0,0 +1,143 @@
+import { ConfigurationError } from "@pipedream/platform";
+import common from "../../common.mjs";
+import utils from "../../common/utils.mjs";
+import constants from "../../constants.mjs";
+
+const { discord } = common.props;
+
+export default {
+  key: "discord_bot-send-message-with-file",
+  name: "Send Message With File",
+  description: "Post a message with an attached file. [See the docs here](https://discord.com/developers/docs/reference#uploading-files)",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    discord,
+    ...common.props,
+    filePath: {
+      type: "string",
+      label: "File Path",
+      description: "The path to the image file saved to the `/tmp` directory (e.g. `/tmp/image.png`). [see docs here](https://pipedream.com/docs/workflows/steps/code/nodejs/working-with-files/#the-tmp-directory).",
+    },
+    message: {
+      propDefinition: [
+        discord,
+        "message",
+      ],
+    },
+    userId: {
+      propDefinition: [
+        discord,
+        "userId",
+        (c) => ({
+          guildId: c.guildId,
+        }),
+      ],
+      description: "Select either an user or a channel",
+      optional: true,
+    },
+    channelId: {
+      propDefinition: [
+        discord,
+        "channelId",
+        ({ guildId }) => ({
+          guildId,
+          notAllowedChannels: constants.NOT_ALLOWED_CHANNELS,
+        }),
+      ],
+      description: "Select either a channel or an user",
+      optional: true,
+    },
+    embeds: {
+      propDefinition: [
+        discord,
+        "embeds",
+      ],
+    },
+    threadID: {
+      propDefinition: [
+        discord,
+        "threadID",
+      ],
+    },
+    username: {
+      propDefinition: [
+        discord,
+        "username",
+      ],
+    },
+    avatarURL: {
+      propDefinition: [
+        discord,
+        "avatarURL",
+      ],
+    },
+    includeSentViaPipedream: {
+      type: "boolean",
+      optional: true,
+      default: true,
+      label: "Include link to workflow",
+      description: "Defaults to `true`, includes a link to this workflow at the end of your Discord message.",
+    },
+  },
+  methods: {
+    ...common.methods,
+    getUserInputProps(omit = [
```

Comment:
It looks like you're not using this method, so you can remove it.

Reply:
done!